sys/picolibc_syscalls_default/syscalls.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2019 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    sys_picolibc PicoLibc system call
+ * @ingroup     sys
+ * @brief       PicoLibc system call
+ * @{
+ *
+ * @file
+ * @brief       PicoLibc system call implementations
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+
+#include "log.h"
+#include "periph/pm.h"
+#include "stdio_base.h"
+
+#ifdef MODULE_CORTEXM_COMMON
+/**
+ * @brief manage the heap
+ */
+extern char _sheap;                 /* start of the heap */
+extern char _eheap;                 /* end of the heap */
+
+char * __heap_start = &_sheap;
+char * __heap_end = &_eheap;
+#else
+/**
+ * @brief manage the heap
+ */
+extern char _heap_start;                 /* start of the heap */
+extern char _heap_end;                 /* end of the heap */
+
+char * __heap_start = &_heap_start;
+char * __heap_end = &_heap_end;
+#endif
```

Comment:
There is probably a better and cleaner way to glue this together.

Reply:
I still think we should just use the same names everywhere #12328

Reply:
That would at least simplify things here

Reply:
#14455 is merged so should be simpler now.

Reply:
Rebased and simplified!